rocksdb/rocksdb-overview.md

```diff
@@ -0,0 +1,61 @@
+
+## TiKV architecture
+
+The architecture of TiKV is illustrated as follows:
+
+![TiKV RocksDB](/media/tikv-rocksdb.png)
+
+As the storage engine of TiKV, RocksDB is used to store Raft logs and user data. All data in a TiKV node shares two RocksDB instances. One is for Raft log (often called raftdb), and the other is for user data and MVCC metadata (often called kvdb). There are four Column Families (CF) in kvdb: raft, lock, default, and write:
+
+* raft CF: Store metadata of each Region. It occupies only a very small amount of space, and users do not need to care.
+* lock CF: Store the pessimistic lock of pessimistic transactions and the Prewrite lock for distributed transactions. After the transaction is committed, the corresponding data in lock CF is deleted quickly. Therefore, the size of data in lock CF is usually very small (less than 1 GB). If the data in lock CF increases a lot, it means that a large number of transactions are waiting to be committed, and that the system meets a bug or failure.
+* write CF: Store the user's real written data and MVCC metadata (the start timestamp and commit timestamp of the transaction to which the data belongs). When the user writes a row of data, it is stored in the write CF if the data length is less than 255 bytes. Otherwise, it is stored in the default CF. In TiDB, the secondary index only occupies the space of write CF, since the value stored in the non-unique index is empty and the value stored in the unique index is the primary key index.
+* default CF: Store data longer than 255 bytes.
+
+## RocksDB memory usage
+
+To improve the reading performance and reduce the reading operations to the disk, RocksDB divides the files stored on the disk into blocks based on a certain size (the default is 64 KB). When reading a block, it first checks if the data already exists in BlockCache in memory. If true, it can read the data directly from memory without accessing the disk.
+
+BlockCache discards the least recently used data according to the LRU algorithm. By default, TiKV devotes 45% of the system memory to BlockCache. Users can also modify the `storage.block-cache.capacity` configuration to an appropriate value by themselves. However, it is not recommended to exceed 60% of the total system memory.
+
+The data written to RocksDB is written to MemTable firstly. When the size of a MemTable exceeds 128 MB, it switches to a new MemTable. There are 2 RocksDB instances in TiKV, a total of 4 ColumnFamily. The size limit of a single MemTable for each ColumnFamily is 128 MB. A maximum of 5 MemTables can exist at the same time; otherwise, the foreground writes will be blocked. The memory occupied by this part is at most 2.5 GB (4 x 5 x 128 MB). It is not recommended to change this limit since it costs less memory.
```

Comment:
`will be blocked` or `is blocked`?

Reply:
`4 ColumnFamily` or `4 Column Families`?

Reply:
> `will be blocked` or `is blocked`?

Either is okay. I'd prefer the present tense, but `will be blocked` is acceptable.

Reply:
> `4 ColumnFamily` or `4 Column Families`?

`4 Column Families`.
Also, you may use `Column Family (CF)` on its first appearance, and refers to it as `CF` afterwards. It saves a lot of trouble.

Reply:
I try to use `Column Families (CFs)` on its first appearance, but I'm not sure if it is ok. https://github.com/pingcap/docs/pull/3084/commits/5727e86618b60e8c3cb7ec10948c8f769511846d

Reply:
LGTM 🎉